frappe/integrations/doctype/nextcloud_settings/nextcloud_settings.py

```diff
@@ -0,0 +1,207 @@
+		except JobTimeoutException:
+			if retry_count < 2:
+				args = {
+					"retry_count": retry_count + 1,
+					"upload_db_backup": False #considering till worker timeout db backup is uploaded
+				}
+				enqueue(self.take_backup_nextcloud, queue='long', timeout=1500, **args)
+		except Exception:
+			if isinstance(error_log, str):
+				error_message = error_log + "\n" + frappe.get_traceback()
+			else:
+				file_and_error = [" - ".join(f) for f in zip(did_not_upload, list(set(error_log)))]
+				error_message = ("\n".join(file_and_error) + "\n" + frappe.get_traceback())
+			send_email(False, "NextCloud", "NextCloud Settings", "send_notifications_to", error_message)
+
+	def backup_to_nextcloud(self, did_not_upload, error_log, upload_db_backup=True,):
+		if not frappe.db:
+			frappe.connect()
+		if upload_db_backup:
+			base_url = self.make_baseurl()
+			if not base_url:
+				did_not_upload.append('Failed')
+				error_log.append('NextCloud URL incorrect')
+				return
+			if self.path_to_upload_folder:
+				url = '{0}{1}'.format(base_url, self.path_to_upload_folder)
+				self.upload_path = True
+			else:
+				url = '{0}{1}'.format(base_url, 'Frappe Backups')
+			session = self.make_session()
+
+			# check if folder exist
+			self.check_for_upload_folder(session, url)
+			self.process_uploading(session, url, did_not_upload, error_log)
+
+	def process_uploading(self, session, url, did_not_upload, error_log):
+
+		db_backup, site_config, public_file_backup, private_file_backup = self.prepare_backup()
+
+		db_response = upload_backup(session, url, db_backup)
+		if db_response == 'Failed':
+			did_not_upload.append(db_backup)
+			error_log.append('Failed while uploading DB')
+
+		site_config_response = upload_backup(session, url, site_config)
+		if site_config_response == 'Failed': 
+			did_not_upload.append(site_config)
+			error_log.append('Failed while uploading Site Config')
+
+		# file backup
+		if cint(self.backup_files) and db_response != 'Failed' and site_config_response != 'Failed':
+			self.file_upload(session, url, public_file_backup, private_file_backup, did_not_upload, error_log)
+
+	def file_upload(self, session, url, public_file_backup, private_file_backup, did_not_upload, error_log):
+		if public_file_backup:
+			response_public_file = upload_backup(session, url, public_file_backup)
+			if response_public_file == 'Failed': 
+				did_not_upload.append(public_file_backup)
+				error_log.append('Failed while uploading Public files')
+		if private_file_backup:
+			response_private_file = upload_backup(session, url, private_file_backup)
+			if response_private_file == 'Failed': 
+				did_not_upload.append(private_file_backup)
+				error_log.append('Failed while uploading Private files')
+
+	def prepare_backup(self, db_backup=None, site_config=None, public_file_backup=None, private_file_backup=None):
+		if frappe.flags.create_new_backup:
+			backup = new_backup(ignore_files= True if cint(self.backup_files) else False)
```

Comment:
I re-wrote this whole function as:

```suggestion
	def prepare_backup(self):
		odb = new_backup(ignore_files=self.backup_files, force=frappe.flags.create_new_backup)
		database, public, private, config = odb.get_recent_backup(older_than=24 * 30)
		return database, config, public, private
```

I don't understand why you processed the file paths to retrieve the file names, only to append it to the site's backup path and generate the same path again. And the reason for the extra unused kwargs. But my suggestion should cover the same functionality too.